### ISSUE TITLE
Remember credentials until page refresh.

### ISSUE THREAD

**Issue by mseashor** [mulesoft/api-console#180]
I would like to default or remember the previous credentials (Basic auth) when a new API request is sent from the api-console.  

Currently, each time a new endpoint is selected in the API console, the credentials need to be updated.  I think 'remembering' the credentials would be ideal, but just allowing 'defaulting' credentials to a specific user/pass would fix 98% of the issue (developers/integrators complaining about typing in credentials for a dev env for every request).

I have figured out how to update the api-console.js javascript (below) to default the credential, but I'm wondering:
1.  Is there's a way to default the credential from the calling HTML file so as to not 'fork' the javascript?
2.  Would an enhancement to 'remember' the credentials (at least until page refresh) be entertained?

Thanks! (And I love the API Console - it makes my life so much easier!)

-Matt

PS: My apologies if there's a better forum for this discussion.  Please inform me if there is.

------------------------
My 'hack' to api-console.js to default user/pass in API Console.  Change is from: $scope.credentials = {}" to $scope.credentials = {username: "user", password: "pass"};

          $scope.showSpinner              = false;
          $scope.securitySchemes          = $scope.methodInfo.securitySchemes();
          $scope.credentials              = {username: "user",password:"pass"};
          $scope.traits                   = $scope.readTraits($scope.methodInfo.is);
          $scope.context.customParameters = { headers: [], queryParameters: [] };

**Comment by mseashor** [comment]
FYI: I created a pull request which illustrates/fixes the issue.  

The end result is you only have to enter your credentials each time the page is loaded, not for every endpoint.

I know it's a small thing, but this change really helps usability of the API console with secured endpoints.  Please let me know if you see issues with this or if there are additional changes (tests, etc) I would need to look at to get this merged in.

**Comment by sichvoge** [comment]
Hi @mseashor, did you add any tests to your PR? If not, please check that your changes do not influence anything else.

**Comment by mseashor** [comment]
Awesome.  I'll run all the tests and make sure they don't break anything.
I'll also try and add a test if I can figure it out (I'm a bit of a
JavaScript noob).

**Comment by sichvoge** [comment]
Brilliant! Let us know when you are done.

**Comment by mseashor** [comment]
@sichvoge - I think this should be ready to go.  See the pull request (https://github.com/mulesoft/api-console/pull/183) for more info.  

*Summary*: Tests all run successfully.  I added a test which tests a RAML file with two resources.  It enters the user/pass on the first and ensures they are the same when switching to the second.

*Note*: I ran into an issue where Travis CI's default resolution was low enough that default 'Try It' functionality would not be displayed (caused errors when running in Travis CI, but works on my computer).  I updated the .travis.yml to increase the resolution, this fixed the issue.  I think this will be helpful to prevent future confusing issues.  The alternative would be to explicitly re-size the browser to a small size so it runs the same all the time.
